docs/markdown/Freedesktop-module.md

```diff
@@ -0,0 +1,112 @@
+  value is not provided then the field is unset.
+
+- `generic_name` is the `GenericName` in the .desktop file. If this value is
+  not provided then the field is unset.
+
+- `icon` is the `Icon` value in the .desktop file. Meson does not validate
+  this value in anyway. If not provided the field is unset.
+
+- `categories` a meson list of categories for this program. These values will
+  be joined into a single semicolon delimited string in the .desktop file.
+  Meson does not validate this value in any way. If not provided the field is
+  unset.
+
+- `install_name` The name of the .desktop file without the file extension. If
+  this is unset the name of the value of `name` or the name of the executable
+  will be used, in that order.
+
+- `mimetype` A meson list of mimetypes that this program handles. These
+  values will be joined into a single semicolon delimited string in the
+  .desktop file. Meson does not validate these values in any way. If not
+  provided the field is unset.
+
+- `dbus` A boolean that populates the `DBusActivatable` field in the .desktop
+file. Defaults to false
+
+- `no_display` A boolean for the `NoDisplay` field.
+
+- `only_show_in` A list of strings to populate the `OnlyShowIn` field.
+
+- `not_show_in` A list of strings to populate the `NotShowIn` field.
+
+- `startup_notify` A Boolean for to populate the `StartupNotify` field.
+
+- `startup_wm_class` A Boolean for to populate the `StartupWMClass` field.
+
+- `implements` A list of strings that will be `;` joined to populate the
+  `Implements` field.
+
+- `keywords` A list of strings that will be `;` joined to populate the
+  `Keywords` field.
+
+- `extra_args` A dict of string to string|list of string|boolean's for
+  extended attributes. The keys must start with `X-`. Lists will be converted
+  in `;` delimited strings. If a value needs to be formatted differently the
+  caller must format it themselves and pass the value as a string. Meson
+  validates only that all keys start with `X-`, it is up to the caller to
+  ensure that these keys are correctly formatted.
+
+- `test_target` A boolean. If this is set to true then a test target to
+  validate the .desktop file is also generated. If this is true then the
+  program `desktop-file-validate` is required
+
+- `actions` A list of dictionaries for additional actions for the .desktop
+  file. If this value is not provided then no additional actions are set.
+
+The actions dictionaries have the following signature:
+
+- `exec` Additional arguments to pass to the Executable. The path to the
+  executable is provided, these are treated as extra arguments. Required.
+
+- `name` the name of the action. Required.
+
+- `icon` Icon for the action, uses the same logic as the main `icon`
+  argument. Optional.
+
+```meson
+fdo = import('freedesktop')
+
+exe = executable(..., gui_app : true, install : true)
+fdo.desktop_file(
+  exe,
+  generic_name : 'IDE',
+  actions : [
+    {
+      name : 'Import',
```

Comment:
keys must be quoted strings here.

Reply:
Unless we introduce intermediary object like `action = fd.action('Import', exec : [...])`